```c
#include <setjmp.h>
#include <stdio.h>

#include "q2.h"

void jump_target(jmp_buf *stack_info) {
    int longjmp_value;
    if ((longjmp_value = setjmp(*stack_info)) == 0) {
        printf("Setup land point with setjmp\n");
    } else {
        printf("Successfully jumped into jump_target. Received value %d\n", longjmp_value);
    }
}

void chpt6_q2() {
    jmp_buf stack_info;
    jump_target(&stack_info);
    longjmp(stack_info, 12345); // Jump into jump_target, but it has already returned.
    // Should segfault!
}
```

Result:

```console
vagrant@vagrant:/vagrant/c$ run chp6 q2
Setup land point with setjmp
Successfully jumped into jump_target. Received value 12345
Segmentation fault (core dumped)
```

The program successfully jumps in this case, but truly this is an undefined behavior.

The segmentation fault occurs as we reach the end of `jump_target` after the jump and there is no record in the stack of a caller for this function (cause it has already returned).

Therefore, `jump_target` probably returns to some random location at the 2nd time and generates a segfault for trying to execute a non-executable page, the pointed intruction being invalid or whatever other reason.
